server/src/main/java/org/opensearch/index/IndexSortConfig.java

```diff
@@ -200,6 +200,7 @@ public boolean hasPrimarySortOnField(String field) {
      * or returns null if this index has no sort.
      */
     public Sort buildIndexSort(
+        boolean shouldWidenIndexSortTpe,
```

Comment:
I think you don't need this parameter, we could get it from `indexSettings` on construction:

```
 public IndexSortConfig(IndexSettings indexSettings)
```

Reply:
Done